src/compiler/tools/results/FileTextChanges.ts

```diff
@@ -59,7 +61,7 @@ export class FileTextChanges {
         if (this._isApplied)
             return;
 
-        let file: SourceFile | undefined;
+        let file = this.getSourceFile();
 
         if (this.isNewFile() && file != null && !options.overwrite) {
```

Comment:
this expression was always false

Reply:
Thanks! I recently broke this.

Reply:
Is there a test for this? We should add one if not.

Reply:
Yes I think there's already tests to verify "should throw if a file text change instruct to create a new file that already exists" . (and overwrite is not given)

Reply:
I just tried reverting the code here and the tests pass. I'll add a test for this.